docs/USING_PRO.md

```diff
@@ -89,9 +89,93 @@ slugger.slug('foo-1') // foo-1-2
 - image(*string* href, *string* title, *string* text)
 - text(*string* text)
 
+<h2 id="tokenizer">The tokenizer</h2>
+
+The tokenizer defines how to turn markdown text into tokens.
+
+**Example:** Overriding default `codespan` tokenizer to include latex.
+
+```js
+// Create reference instance
+const marked = require('marked');
+
+// Get reference
+const tokenizer = new marked.Tokenizer();
+const originalCodespan = tokenizer.codespan;
+// Override function
+tokenizer.codespan = function(src) {
+  const match = src.match(/\$+([^\$\n]+?)\$+/);
+  if (match) {
+    return {
+      type: 'codespan',
+      raw: match[0],
+      text: match[1].trim()
+    };
+  }
+  return originalCodespan.apply(this, arguments);
+};
+
+// Run marked
+console.log(marked('$ latex code $', { tokenizer }));
```

Comment:
Nice example!

Reply:
Can we also achieve a dynamic TOC based on headings using a custom tokenizer?

Reply:
It seems like it would be easier to use the Renderer for that. [example](https://github.com/markedjs/marked/issues/545#issuecomment-74505539)

Reply:
Oh yeah. So then couldn't this LaTeX example also be achieved using the Renderer instead of Tokenizer?

Reply:
No because the dollar sign (`$`) isn't a valid code token starter normally so it would just be counted as text.

I'm sure the example code would fail on edge cases but it is more about showing how to extend the Tokenizer than create a robust LaTeX interpreter.